### PR TITLE
Correction Exception HttpClient Timeout

### PR DESCRIPTION
**Pull request by vincequeiroz**
<strong>Exception:</strong>
<pre>
root@vagrant-ubuntu-trusty-64:/vagrant/erp/source/deploy# php public/index.php geoip download
Downloading http://geolite.maxmind.com/download/geoip/database/GeoLiteCity.dat.gz...
======================================================================
   The application has thrown an exception!
======================================================================
 Zend\Http\Client\Adapter\Exception\RuntimeException
 Error in cURL request: Operation timed out after 9999 milliseconds with 2548269 out of 11894972 bytes received
----------------------------------------------------------------------
/vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/Http/Client/Adapter/Curl.php:422
#0 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/Http/Client.php(1379): Zend\Http\Client\Adapter\Curl->write('GET', Object(Zend\Uri\Http), '1.1', Array, '')
#1 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/Http/Client.php(890): Zend\Http\Client->doRequest(Object(Zend\Uri\Http), 'GET', false, Array, '')
#2 /vagrant/erp/source/deploy/vendor/snapshotpl/zf-snap-geoip/src/ZfSnapGeoip/Controller/ConsoleController.php(127): Zend\Http\Client->send()
#3 /vagrant/erp/source/deploy/vendor/snapshotpl/zf-snap-geoip/src/ZfSnapGeoip/Controller/ConsoleController.php(91): ZfSnapGeoip\Controller\ConsoleController->getDbResponse()
#4 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/Mvc/Controller/AbstractActionController.php(83): ZfSnapGeoip\Controller\ConsoleController->downloadAction()
#5 [internal function]: Zend\Mvc\Controller\AbstractActionController->onDispatch(Object(Zend\Mvc\MvcEvent))
#6 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(468): call_user_func(Array, Object(Zend\Mvc\MvcEvent))
#7 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(207): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#8 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/Mvc/Controller/AbstractController.php(116): Zend\EventManager\EventManager->trigger('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#9 /vagrant/erp/source/deploy/vendor/snapshotpl/zf-snap-geoip/src/ZfSnapGeoip/Controller/ConsoleController.php(73): Zend\Mvc\Controller\AbstractController->dispatch(Object(Zend\Console\Request), Object(Zend\Console\Response))
#10 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/Mvc/DispatchListener.php(113): ZfSnapGeoip\Controller\ConsoleController->dispatch(Object(Zend\Console\Request), Object(Zend\Console\Response))
#11 [internal function]: Zend\Mvc\DispatchListener->onDispatch(Object(Zend\Mvc\MvcEvent))
#12 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(468): call_user_func(Array, Object(Zend\Mvc\MvcEvent))
#13 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/EventManager/EventManager.php(207): Zend\EventManager\EventManager->triggerListeners('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#14 /vagrant/erp/source/deploy/vendor/zendframework/zendframework/library/Zend/Mvc/Application.php(313): Zend\EventManager\EventManager->trigger('dispatch', Object(Zend\Mvc\MvcEvent), Object(Closure))
#15 /vagrant/erp/source/deploy/public/index.php(15): Zend\Mvc\Application->run()
#16 {main}
======================================================================
   Previous Exception(s):
</pre>

<strong>Error with the try catch:</strong>
<pre>
root@vagrant-ubuntu-trusty-64:/vagrant/erp/source/deploy# php public/index.php geoip download
Downloading http://geolite.maxmind.com/download/geoip/database/GeoLiteCity.dat.gz...
Error in cURL request: Resolving timed out after 1514 milliseconds
Error during file download occured
</pre>

<strong>Result with HttpClient timeout option with 30 seconds.</strong>
<pre>
root@vagrant-ubuntu-trusty-64:/vagrant/erp/source/deploy# php public/index.php geoip download
Downloading http://geolite.maxmind.com/download/geoip/database/GeoLiteCity.dat.gz...
Download completed
Unzip the downloading data...
Unzip completed (/vagrant/erp/source/deploy/vendor/snapshotpl/zf-snap-geoip/config/../data/GeoLiteCity.dat)
</pre>

### CODE REVIEW

File: src/ZfSnapGeoip/Controller/ConsoleController.php [hunk 1]
```diff
@@ -121,10 +121,17 @@ public function getDbResponse()
 
         $this->writeLine(sprintf('Downloading %s...', $source), Color::LIGHT_YELLOW);
 
-        $this->httpClient->setUri($source);
-        $this->httpClient->setMethod(Request::METHOD_GET);
+        try {
+            $this->httpClient->setUri($source);
+            $this->httpClient->setMethod(Request::METHOD_GET);
+            $this->httpClient->setOptions(array('timeout' => 30));
```
Comment (snapshotpl) [inline src/ZfSnapGeoip/Controller/ConsoleController.php]: Can you close all options in configuration file?

File: src/ZfSnapGeoip/Controller/ConsoleController.php [hunk 2]
```diff
@@ -121,10 +121,17 @@ public function getDbResponse()
 
         $this->writeLine(sprintf('Downloading %s...', $source), Color::LIGHT_YELLOW);
 
-        $this->httpClient->setUri($source);
-        $this->httpClient->setMethod(Request::METHOD_GET);
+        try {
```
Comment (snapshotpl) [inline src/ZfSnapGeoip/Controller/ConsoleController.php]: Use try direct before send() method

File: src/ZfSnapGeoip/Controller/ConsoleController.php [hunk 3]
```diff
@@ -121,10 +121,17 @@ public function getDbResponse()
...
+            $this->httpClient->setUri($source);
+            $this->httpClient->setMethod(Request::METHOD_GET);
+            $this->httpClient->setOptions(array('timeout' => 30));
+            
+            return $this->httpClient->send();
+        } catch (\Exception $e) {
```
Comment (snapshotpl) [inline src/ZfSnapGeoip/Controller/ConsoleController.php]: Catch Zend\Http exception

File: src/ZfSnapGeoip/Controller/ConsoleController.php [hunk 4]
```diff
@@ -121,10 +121,17 @@ public function getDbResponse()
...
+            $this->writeLine(sprintf('%s', $e->getMessage()), Color::WHITE, Color::RED);
 
-        return $this->httpClient->send();
+            return null;
```
Comment (snapshotpl) [inline src/ZfSnapGeoip/Controller/ConsoleController.php]: Without null

File: config/module.config.php [hunk 5]
```diff
@@ -9,6 +9,11 @@
             'flag' => GEOIP_STANDARD,
             'regionvars' => __DIR__ . '/../../../geoip/geoip/src/geoipregionvars.php',
         ),
+        'http_client' => array(
+            'options' => array(
+                'timeout' => 300
```
Comment (vincequeiroz) [inline config/module.config.php]: I don't believe be necessary another param of config.
http://www.zendframework.com/manual/current/en/modules/zend.http.client.html#configuration
http://www.zendframework.com/manual/current/en/modules/zend.http.client.adapters.html#the-curl-adapter

**Comment by snapshotpl** [comment]
Thanks @vincequeiroz ! I will review it this evening.

**Comment by snapshotpl** [comment]
Thanks @vincequeiroz !

**Pull request merged**